### ISSUE TITLE
Support lists with predetermined sort order

### ISSUE THREAD

**Issue by kirkholloway** [atlassian/react-beautiful-dnd#973]
## Bug or feature request?

Feature request

## Feature request

Support ordered Droppables (cards inside sorted alphabetically, by date, etc.).

Currently this works fairly well. But since the visually indicated place for the card is under wherever it's being held, the cards jump around strangely when going into the sorted order.

I've been thinking about APIs that could enable this. Users could define the indexes a Draggable would be sorted into for each list, and set those from `onDragStart`. Possibly using a callback with an API similar to this:
```
setDroppableSortIndex({
  'droppable-id1': 4,
  'droppable-id2': 2,
  'droppable-id3': 5,
  'droppable-id4': 12,
  ...
});
```
 
A Codepen showing 2 lists sorted numerically and the current behaviour that happens when you drag between two sorted lists:
https://codesandbox.io/s/vqlxj911x3

**Comment by alexreardon** [comment]
You are welcome to sort items however you like! You just need to make sure that the `index` prop of a `Draggable` matches it's visible order. So you would need to be aware in your `onDragEnd` function how the list is sorted when doing your reordering

**Comment by kirkholloway** [comment]
Hey @alexreardon, thanks for replying so quickly! Also, this library is very pretty 😍 

The purpose of the additional API I'm suggesting would be to indicate where the `Draggable` would be positioned once released. Currently doesn't quite look right when the card reorders itself after releasing into a sorted column. If you look at the codepen I added, after releasing the card, there's a slightly jarring reshuffle. I've added colours to the cards from the previously included codesandbox to hopefully demonstrate things a bit better 😁 https://codesandbox.io/s/30p83jmkw1

It would be amazing if we could have the list create an opening for a card at it's sorted index rather than where it's being hovered, and animate it moving into the right position. 😁
